sw/device/silicon_creator/lib/drivers/otbn.c

```diff
@@ -59,6 +59,17 @@ static rom_error_t check_offset_len(uint32_t offset_bytes, size_t num_words,
   return kErrorOk;
 }
 
+/**
```

Comment:
We already have `otbn_is_busy()`. WDYT about leaving the driver as is and adding
```
while(otbn_is_busy()) {
}
```
before we start using OTBN.

Otherwise we will be checking both before and after a command, which is a bit redundant.

Reply:
Sure, that seems fine. Also probably worth adding a basic check that makes sure OTBN isn't locked -- that way by process of elimination it must be idle. And an error is better than an infinite spin in the locked case (although I think OTBN should not be locked here unless it detects an FI attack).

Reply:
Updated the PR to add the `otbn_is_busy()` loop and the locked check to `otbn_cmd_run`, `otbn_write`, and `otbn_dmem_read` (since no writes/reads are permitted in non-idle states). I also added some unit tests for locked states.

Reply:
> OTBN should not be locked here unless it detects an FI attack

Thanks for updating the PR. I agree, I was thinking about the same thing. It looks like the driver also needs something similar to `otbn_busy_wait_for_done()` which is currently in `otbn_util.c`. WDYT about replacing `otbn_is_busy()` with something like:
```
rom_error_t otbn_busy_wait_for_done(void) {
    uint32_t status = UINT32_MAX;
    rom_error_t res = kErrorOk ^ status;
    do {
        status = abs_mmio_read32(kBase + OTBN_STATUS_REG_OFFSET);
    } while (launder(status) != kOtbnStatusIdle && launder(status) != kOtbnStatusLocked);
    res ^= ~status;
    if (launder(res) == kErrorOk) {
        HARDENED_CHECK_EQ(res, kErrorOk);
        HARDENED_CHECK_EQ(abs_mmio_read32(kBase + OTBN_STATUS_REG_OFFSET), kOtbnStatusIdle);
        return res;
    }
    return kOtbnUnavailable;
}
```

If this makes sense, we can update the call sites that use `otbn_is_busy()` and #13931.

Reply:
I've replaced `otbn_is_busy()` with this code, removed the blocking parts from `otbn_write` and `otbn_cmd_run`, and added an `otbn_busy_wait_for_done()` step in `otbn_load_app` (so it runs before mask ROM loads the modexp application but not for subsequent operations).

Just to check that this is the intended outcome: if OTBN is locked, this will produce a hardening check failure. Since OTBN being locked means we should suspect a fault injection attack, this makes sense to me.